src/main/java/org/generationcp/middleware/dao/GermplasmSearchDAO.java

```diff
@@ -1541,32 +1574,10 @@ private boolean addPreFilteredGids(final GermplasmSearchRequest germplasmSearchR
 			.list();
 	}
 
-	public List<UserDefinedField> getGermplasmAttributeTypes(final GermplasmSearchRequest germplasmSearchRequest) {
-		try {
-
-			final List<Integer> gids = this.retrieveSearchGids(germplasmSearchRequest, null, null);
-			final String sql = "select distinct {u.*} from atributs a inner join udflds u "
-				+ " where a.atype = u.fldno"
-				+ " and a.gid in (:gids)"
-				+ " order by u.fname";
-
-			final SQLQuery query = this.getSession().createSQLQuery(sql);
-			query.addEntity("u", UserDefinedField.class);
-			query.setParameterList("gids", gids);
-
-			return query.list();
-		} catch (final HibernateException e) {
-			final String message =
-				"Error with getGermplasmAttributeTypes(GermplasmSearchRequest=" + germplasmSearchRequest + ") : " + e.getMessage();
-			GermplasmSearchDAO.LOG.error(message, e);
-			throw new MiddlewareQueryException(message, e);
-		}
-	}
-
-	public List<UserDefinedField> getGermplasmNameTypes(final GermplasmSearchRequest germplasmSearchRequest) {
+	public List<UserDefinedField> getGermplasmNameTypes(final GermplasmSearchRequest germplasmSearchRequest, final String programUUID) {
```

Comment:
Why is programUUID relevant in getting name types? To my understanding, programUUID is used for variable alias which only applies to attribute/passport variables.

Reply:
@darla-leafnode 
It is necessary when the filter by attributes used.

Reply:
@cuenyad maybe we can rename the function to getGermplasmSearchNameTypes to be more specific?